GL-385: Create the first two steps of the Check Moving Requirement user journey

alexdesi opened the pull request:
### Jira link
https://transformuk.atlassian.net/browse/GL-385
https://transformuk.atlassian.net/browse/GL-388

Create the first two steps of the Check Moving Requirement feature

- Step1 is the /check_moving_requirements/start page
- Step2 is the /check_moving_requirements/edit page

### What?
These pages are part of the user journey to check for the requirements for moving goods from GB to NI

### Why?
Part of the GL job

### Have you? (optional)

- [] Reviewed view changes with stakeholders
WIP

- [ ] Validated mobile responsive behaviour of view changes
WIP

### Deployment risks (optional)
TODO: need to put this feature behind feature flag

alexdesi committed:
GL-385: Create the first two steps of the Check Moving Requirement feature.
Step1 is the /check_moving_requirements/start page
Step2 is the /check_moving_requirements/edit page

alexdesi committed:
GL-385: Add validations for check_moving_requirements_form
Improve error messages on locales/en.yml
Create new initial page /result.html.erb
Added missing specs for check_moving_requirements_controller

jebw commented on app/views/check_moving_requirements/edit.html.erb:
```diff
@@ -0,0 +1,65 @@
+      <%= f.govuk_text_field :commodity_code, width: 'one-quarter',
+                  label: { text: 'Commodity Code'} %>
+
+      <hr class="govuk-section-break govuk-section-break--m govuk-section-break--visible">
```
does this need to self close? not quite sure what does and does not anymore post html5

alexdesi replied:
yes, <hr> is the correct way in html5, XHTML is more restrict and it need to be close

jebw commented on app/views/check_moving_requirements/edit.html.erb:
```diff
@@ -0,0 +1,65 @@
+      </p>
+
+      <%= f.govuk_text_field :commodity_code, width: 'one-quarter',
+                  label: { text: 'Commodity Code'} %>
```
indentation looks a bit off?

jebw commented on app/views/check_moving_requirements/edit.html.erb:
```diff
@@ -0,0 +1,65 @@
+      <%= f.govuk_date_field :moving_date,
+        maxlength_enabled: true,
+        legend: { text: 'When is the movement of goods taking place?', size: 's', tag: 'h1' },
+        hint: { text: "Arrangements can change over time. If you are unsure of the date of movement and want to check arrangements currently in place, enter today's date." } %>
```
Indentation?

jebw commented on app/controllers/check_moving_requirements_controller.rb:
```diff
@@ -0,0 +1,33 @@
+class CheckMovingRequirementsController < ApplicationController
```
Can this be namespaced under `GreenLanes`, thanks

alexdesi replied:
I agree, done!

alexdesi committed:
GL-385: Add feature flag to enable/disable check moving requirement pages

jebw commented on .env.development:
```diff
@@ -10,6 +10,7 @@ BASIC_USERNAME=test
 BETA_SEARCH_HEADING_STATISTICS_THRESHOLD=3
 BETA_SEARCH_FACET_FILTER_DISPLAY_PERCENTAGE_THRESHOLD=5
 BETA_SEARCH_SWITCHING_ENABLED=true
+CHECK_MOVING_REQUIREMENTS_ENABLED=false
```
GREEN_LANES_ENABLED

alexdesi replied:
Done, also I reused/renamed the  ALLOW_GREEN_LANES (added by @rasikasri )

jebw commented on config/routes.rb:
```diff
@@ -215,6 +215,16 @@
+  # resources :check_moving_requirements, only: %i[show update edit] do
+  #   get 'start', on: :collection
+  # end
+
```
I know I'd originally suggested a singular resource but it feels like this set of screens maps better to a plural resource?

```ruby
  scope module: :green_lanes do
    resources :check_moving_goods_requirements, only: %i[index edit update destroy]
  end

alexdesi replied:
I adopted a similar solution, under the existing
`    namespace :green_lanes do`

jebw reviewed (changes requested):
Hey there, looks good - there's some changes which I think would benefit it

1. Namespacing the code (models/conrtollers/views) - even if the routes need to be at the top level, we can still keep the code a bit more organised
2. I think its possible to use `resources ...` in the `config/routes.rb` to clean that up a little
3. It might make sense to have a generic green lanes feature flag I think since we'll want to use the same for the results screens _but_ if you're envisioning those just using this feature flag then that should be fine

alexdesi committed:
GL-385: Improve routes for check_moving_requirements

alexdesi committed:
GL-385: Rename Green Lanes feature flag.
This PR renames and reuses the existing feature flag (to follow the current convention)
ALLOW_GREEN_LANE -> GREEN_LANE_ENABLED

alexdesi committed:
GL-385: move check_moving_requirements into GreenLane module

alexdesi committed:
GL-385: Fix string too long.

jebw commented on app/forms/check_moving_requirements_form.rb:
```diff
@@ -0,0 +1,13 @@
+class CheckMovingRequirementsForm
```
Namespacing

alexdesi committed:
Fix a terraform variable

alexdesi committed:
GL-385: refactor: move check_moving_requirements_form under green_lanes module

jebw reviewed (dismissed):
LGTM

alexdesi committed:
GL-385: Various design improvement on check_moving_requirements/start and check_moving_requirements/edit pages.

alexdesi committed:
gl-385: Disable search form for green_lanes pages.

alexdesi committed:
GL-385: Fix brocken tests in check_moving_requirements_controller_spec.

alexdesi committed:
Add only numbers validation for the commodity code.

alexdesi committed:
Fix a terraform var name.
